web/src/input_pill.ts

```diff
@@ -346,7 +346,8 @@ export function create<T>(opts: InputPillCreateOptions<T>): InputPillContainer<T
             if (
                 selection?.type !== "range" &&
                 e.key === "Backspace" &&
-                (funcs.value(e.target).length === 0 || selection?.anchorOffset === 0)
+                (funcs.value(e.target).length === 0 ||
+                    (selection?.anchorOffset === 0 && selection?.toString()?.length === 0))
```

Comment:
Why is `selection?.type !== "range"` in the test case in question? https://developer.mozilla.org/en-US/docs/Web/API/Selection/type seems to suggest that this length should always be 0 already.

Reply:
@timabbott I looked up on the internet regarding this, and found this StackOverflow [answer](https://stackoverflow.com/questions/5072155/how-do-i-get-window-getselection-to-work-for-an-input-type-text-field), which states that in most browsers, ```selection?.type !== "range"``` only works with text nodes and elements within the document and not to ```<input>``` and ```<textarea>```.
However, ```window.getSelection().toString()``` returns the selected text within a focussed text input or textarea. 

So I replaced ```selection?.type !== "range"``` with ```selection?.toString()?.length === 0``` and found out it removes the bug.
Hence, I am force pushing the same.

Reply:
OK, but I guess both of these are `div class="input"` contenteditable elements.

Can you test this PR in Firefox; the StackOverflow answer mentions "webkit browsers", and that seems worth verifying.

Reply:
@timabbott Yes, the PR is working as expected in Firefox as well.